openpathsampling/ensemble.py

```diff
@@ -165,13 +166,15 @@ def check(self, trajectory=None, reset=None):
             if self.debug_enabled:
                 logger.debug("Resetting cache " + str(self))
             if self.direction > 0:
-                self.start_frame = trajectory.get_as_proxy(0)
-                self.prev_last_frame = trajectory.get_as_proxy(-1)
+                # TODO: this can be hit with trajectory is None?
```

Comment:
If `trajectory is None` the `else statement on line `159` is hit, which means that if `self.direction` is set this will error out (this was already the case in the old code), the same is true for line `174`

Reply:
But this would always error out if it was ever hit with `Trajectory=None`? 
It will now error out with `NameError: name 'get_frame' is not defined` but in the old code it should have error out with `AttributeError: 'NoneType' has no attribute 'get_as_proxy'`

Reply:
Yeah, I think it's fine. I think that more correctly, I shouldn't have allowed `trajectory` to be `None`. I don't know what motivated that -- maybe in early iterations of the caching, I didn't want it to raise errors if it wasn't called with a trajectory.